File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/MachineDetail.java
package com.pennapps.labs.pennmobile.classes;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Jackie on 2017-12-07.
 */

public class MachineDetail implements Comparable<MachineDetail> {

    @SerializedName("id")
    @Expose
    private int id;

    @SerializedName("status")
    @Expose
    private String status;

    @SerializedName("time_remaining")
    @Expose
    private int timeRemaining;

    @SerializedName("type")
    @Expose
    private String type;

    public int getId() {
        return id;
    }

    public String getStatus() {
        return status;
    }

    public int getTimeRemaining() {
        return timeRemaining;
    }

    public void setTimeRemaining(int time) {
        this.timeRemaining = time;
    }

    public String getType() {
        return type;
    }

    @Override
    public int compareTo(MachineDetail machineDetail) {

        if (timeRemaining == machineDetail.getTimeRemaining()) {
            return 0;
        } else if (timeRemaining > machineDetail.getTimeRemaining()) {
            return 1;
        } else {
            return -1;
        }
    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/MachineDetail.kt
package com.pennapps.labs.pennmobile.classes

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Varun on 10/26/2018.
 */

class MachineDetail : Comparable<MachineDetail> {

    @SerializedName("id")
    @Expose
    val id: Int = 0

    @SerializedName("status")
    @Expose
    val status: String? = null

    @SerializedName("time_remaining")
    @Expose
    var timeRemaining: Int = 0

    @SerializedName("type")
    @Expose
    val type: String? = null

    override fun compareTo(machineDetail: MachineDetail): Int {

        return if (timeRemaining == machineDetail.timeRemaining) {
            0
        } else if (timeRemaining > machineDetail.timeRemaining) {
            1
        } else {
            -1
        }
    }
}